### PR TITLE
Add Mono<T>.await

### LINKED ISSUE #1587: Add await() extension on Rector Mono type
It should simply be an inline alias to `Publisher.awaitSingle()`. But special care shall be taken about `Mono<Void>` types that never emit any value. They need `awaitCompletion()` which is missing altogether now.

### PR DESCRIPTION
**Pull request by dkhalanskyjb**
Fixes #1587

### CODE REVIEW

File: reactive/kotlinx-coroutines-reactor/src/Mono.kt [hunk 1]
```diff
@@ -34,6 +32,17 @@ public fun <T> mono(
...
+/**
+ * Awaits the completion of the [Mono] without blocking the thread.
+ * Returns the resulting value, or `null` if no value is produced, or throws the corresponding exception if this
+ * [Mono] has produced an error.
+ *
+ * This suspending function is cancellable.
+ * If the [Job] of the current coroutine is cancelled or completed while this suspending function is waiting, this
+ * function immediately resumes with [CancellationException] and cancels its subscription.
+ */
+public suspend fun <T> Mono<T>.await(): T? = awaitFirstOrNull()
```
Comment (qwwdfsad) [inline reactive/kotlinx-coroutines-reactor/src/Mono.kt:44]: It seems like the `awaitOrNull` can be a more appropriate name.
We can discuss it along with `singleOr*` deprecation if you don't mind

**Review by qwwdfsad — CHANGES_REQUESTED** [review]

File: reactive/kotlinx-coroutines-reactor/src/Mono.kt (hunk 1 above)
Comment (qwwdfsad) [reply reactive/kotlinx-coroutines-reactor/src/Mono.kt:44]: Or, alternatively, `awaitCompletion`?

Comment (dkhalanskyjb) [reply reactive/kotlinx-coroutines-reactor/src/Mono.kt:44]: Wouldn't we have to rename the existing `MaybeSource<T>.await()` then?

Comment (qwwdfsad) [reply reactive/kotlinx-coroutines-reactor/src/Mono.kt:44]: Good point, I'll revisit them

**Comment by dkhalanskyjb** [comment]
Closed in favor of https://github.com/Kotlin/kotlinx.coroutines/pull/2628

**Pull request closed without merging**